### PR TITLE
PWGHF: adding THnSparse for BDT and cutVariation to Xic task

### PR DESCRIPTION
**Pull request by cterrevo**

### CODE REVIEW

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 1]
```diff
@@ -339,6 +431,39 @@ struct HfTaskXic {
...
+            if constexpr (useMl) {
+              if (candidate.mlProbXicToPKPi().size() == 3) {
+                outputBkg = candidate.mlProbXicToPKPi()[0];    /// bkg score
+                outputPrompt = candidate.mlProbXicToPKPi()[1]; /// prompt score
+                outputFD = candidate.mlProbXicToPKPi()[2];     /// non-prompt score
+              }
+              /// Fill the ML outputScores and variables of candidate (todo: add multiplicity)
+              registry.get<THnSparse>(HIST("hnXicVarsWithBdt"))->Fill(massXic, ptCandidate, outputBkg, outputPrompt, outputFD, candidate.originMcRec());
+            } else {
+              registry.get<THnSparse>(HIST("hnXicVars"))->Fill(massXic, ptCandidate, candidate.ptProng0(), candidate.ptProng1(), candidate.ptProng2(), candidate.chi2PCA(), candidate.decayLength(), candidate.decayLengthXY(), candidate.cpa(), candidate.originMcRec());
```
Comment (fcatalan92) [inline PWGHF/D2H/Tasks/taskXic.cxx:449]:
```suggestion
              registry.get<THnSparse>(HIST("hnXicVars"))->Fill(massXic, ptCandidate, candidate.chi2PCA(), candidate.decayLength(), candidate.decayLengthXY(), candidate.cpa(), candidate.originMcRec());
```
The prong pT is not in the `THnSparse` axes

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 2]
```diff
@@ -339,6 +431,39 @@ struct HfTaskXic {
...
+            if constexpr (useMl) {
+              if (candidate.mlProbXicToPiKP().size() == 3) {
+                outputBkg = candidate.mlProbXicToPiKP()[0];    /// bkg score
+                outputPrompt = candidate.mlProbXicToPiKP()[1]; /// prompt score
+                outputFD = candidate.mlProbXicToPiKP()[2];     /// non-prompt score
+              }
+              /// Fill the ML outputScores and variables of candidate (todo: add multiplicity)
+              registry.get<THnSparse>(HIST("hnXicVarsWithBdt"))->Fill(massXic, ptCandidate, outputBkg, outputPrompt, outputFD, candidate.originMcRec());
+            } else {
+              registry.get<THnSparse>(HIST("hnXicVars"))->Fill(massXic, ptCandidate, candidate.ptProng0(), candidate.ptProng1(), candidate.ptProng2(), candidate.chi2PCA(), candidate.decayLength(), candidate.decayLengthXY(), candidate.cpa(), candidate.originMcRec());
```
Comment (fcatalan92) [inline PWGHF/D2H/Tasks/taskXic.cxx:463]:
```suggestion
              registry.get<THnSparse>(HIST("hnXicVars"))->Fill(massXic, ptCandidate, candidate.chi2PCA(), candidate.decayLength(), candidate.decayLengthXY(), candidate.cpa(), candidate.originMcRec());
```

**Review by fcatalan92 — CHANGES_REQUESTED** [review]
Hi @cterrevo, I have just a few comments. Otherwise, the PR is fine to me.

**Comment by cterrevo** [comment]
> Hi @cterrevo, I have just a few comments. Otherwise, this PR is fine to me!

Thanks Fabio! the fixes are implemented, cheers, 
Cristina

**Review by fcatalan92 — APPROVED** [review]

**Pull request merged**